tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinatorHelpers.Core.cs

```diff
@@ -78,7 +80,17 @@ internal static void CheckPathParams(this Security security, HttpContext context
             {
                 var securityCoordinator = SecurityCoordinator.Get(security, span, transport);
                 var args = new Dictionary<string, object> { { AddressesConstants.RequestPathParams, pathParams } };
-                var result = securityCoordinator.RunWaf(args);
+                IResult? result;
+                // we need these tests for netcoreapp3.1 and lower, as otherwise it throws
+                if (context.Features.Get<ISessionFeature>() is not null && (context.Session?.IsAvailable ?? false))
```

Comment:
Just FYI, this isn't just about whether it's enabled, it's also about whether the session middleware has executed yet. I don't think that matters here, as you're not running this until the endpoint executes, right?

Also, minor nit, but `context.Session` just calls `context.Features.Get<ISessionFeature>()` again, so you may as well reuse the value instead of calling it again I think:
```suggestion
                if (context.Features.Get<ISessionFeature>() is { } feature && (feature.Session?.IsAvailable ?? false))
```

Reply:
Good point thank you! yes it's only called `OnRoutingEndpointMatched`, so when it's available , changed in  1c80b179fa7ad846fc7f9c46eb5d7f71d1009e18